sdk/python/kfp/dsl/executor.py

```diff
@@ -276,7 +277,7 @@ def write_executor_output(self,
             runtime_artifact = {
                 'name': artifact.name,
                 'uri': artifact.uri,
-                'metadata': artifact.metadata,
+                'metadata': self.__check_inf_and_nan_values(artifact.metadata),
```

Comment:
Does this have any issues with downstream serialization/deserialization on the supporting backends?

Reply:
It should not as it basically scans the Python structure for all `inf` and `nan` values and converts to protojson supported values.

Reply:
I'm curious about what happens downstream of that. Does ML Metadata know how to handle these values? Does the UI display them correctly? If not, is the task execution still considered successful?

Just want to make sure this change is compatible with the rest of the stack.

Reply:
Both ML Metadata will handle these values and UI will display them correctly.

Reply:
@connor-mccarthy Any updates?

Reply:
@rimolive, these values are not handled/displayed correctly on the Vertex Pipelines UI.

I think a better approach for us is to update the sample, rather than add special handling for this artifact type and make the sample work. JSON does not support infinity/nan, per [RFC 4627 Section 2.4](https://web.archive.org/web/20160414190115/http://tools.ietf.org/html/rfc4627#section-2.4). Notice the red highlights on the GitHub UI.

![image](https://github.com/kubeflow/pipelines/assets/55268212/e0d9fdf0-d6db-40fb-9e9f-4f08a3ff6a19)

Adding special serialization logic for infinity/nan adds additional contracts throughout the pipelines system, requiring that all consumers of this JSON implement custom JSON deserializers with matching logic. This is not desirable.

Reply:
I'll tell you once more: This is not a JSON issue, but a protobuf3 issue since [protobuf3](https://protobuf.dev/programming-guides/proto3/#json) supports these values. I agree that JSON does not support infinity/nan, but Python and protobuf3 supports. Fixing the sample will fix only one special case but there will be others.

I'll close this PR and fix the sample. We can revisit this solution if users find other cases related.